### PR TITLE
created custom firebase hooks and refactored all files where firebase…

### PR DESCRIPTION
**Pull request by SteveLorentzen**
… is used

### CODE REVIEW

File: src/components/AddExisitingStudentsModal.tsx [hunk 1]
```diff
@@ -10,31 +10,30 @@ import {
...
 } from '@chakra-ui/react'
-import { IStudent, IStudentInStudentGroup, IStudentGroup, IStudentToAdd, IStudentFact } from 'interfacesAndTypes'
+import { IStudentToAdd, IStudentFact } from 'interfacesAndTypes'
 import StudentPreview from 'components/StudentPreview'
-import firebase from 'firebase'
 import { useFirestore } from 'reactfire'
+import {
+  useStudentGroup,
+  useStudentsInThisStudentGroup,
+  useStudents,
```
Comment (trevorfehrman) [inline src/components/AddExisitingStudentsModal.tsx:19]: https://lh3.googleusercontent.com/proxy/cZW6q3Zdez-Z0BpPmX1uwVevCog8Ncm9dEE8mfvowvJwgpn8x-e58tKNK51UmqlgkM6RWTFiZYik6o1oIb7BUHRTTLzjqw

**Review by trevorfehrman — APPROVED** [review]

File: src/components/AddExisitingStudentsModal.tsx (hunk 1 above)
Comment (SteveLorentzen) [reply src/components/AddExisitingStudentsModal.tsx:19]: I can't make this link work

"403. That’s an error.

Your client does not have permission to get URL /proxy/cZW6q3Zdez-Z0BpPmX1uwVevCog8Ncm9dEE8mfvowvJwgpn8x-e58tKNK51UmqlgkM6RWTFiZYik6o1oIb7BUHRTTLzjqw from this server. (Client IP address: 174.85.32.129)
"

Comment (trevorfehrman) [reply src/components/AddExisitingStudentsModal.tsx:19]: it's just a link to this face:

ಠ_ಠ

Comment (trevorfehrman) [reply src/components/AddExisitingStudentsModal.tsx:19]: like as in, `useStudents` ಠ_ಠ

Comment (trevorfehrman) [reply src/components/AddExisitingStudentsModal.tsx:19]: like as in `useStudents` disapproveFace

Comment (trevorfehrman) [reply src/components/AddExisitingStudentsModal.tsx:19]: carry on

Comment (SteveLorentzen) [reply src/components/AddExisitingStudentsModal.tsx:19]: Did you notice my interface names for those return values?  Like IUseStudentsInGroups?

**Pull request merged**